### PR TITLE
571 cleanup general simulations

### LINKED ISSUE #571: Clean up `GeneralSimulations` class
See https://github.com/Roche/crmPack/blob/main/R/Simulations-class.R#L22 and corresponding class `GeneralSimulations`

To do:
- [x] Clean up similarly as the other classes above in the file
    - [x] Add for simpler navigation in RStudio the lines with `# ClassName ----` and `## class ----` and `## constructor ----`
	- [x] includes changing to `snake_case` where it helps: not needed here
	- [x] documentation cleanup:
    	- [x] replace double `##` with `#`
    	- [x] remove `@keywords class` line
    	- [x] replace `\code{\linkS4class{ClassName}}` with ``[`ClassName`]``
    	- [x] Use ``@slot model (`type or class`)\cr`` for documenting the type or class of the slots
    	- [x] Use simple title `ClassName` followed by the lifecycle badge, followed by the description of the class
    	- [x] Add the `@aliases` line for the class definition
	- [x] Use the argument names for `setClass` and replace `representation()` with simple `c()`
	- [x] remove the `validObject` line
	- [x] user constructor:
		- [x] Move the example here, and for consistency rename the file to start with upper case letter, e.g. `Design` instead of `design`
		- [x] Use `@rdname ClassName-class` to put the constructor documentation in the same place as the class definition
		- [x] Use ``@param model (`type or class`)\cr see slot definition.`` for documenting the arguments
		- [x] Remove `@keywords classes` line
		- [x] Remove `@return` line for now 
- [x] Clean up the example code file
- [x] Add unit tests to https://github.com/Roche/crmPack/blob/main/tests/testthat/test-Simulations-class.R
    - [x] compare other tests in other files and use the same approach

### PR DESCRIPTION
**Pull request by pengguanya**
# Pull Request

Fixes #571

### CODE REVIEW

**Comment by pengguanya** [comment]
@danielinteractive Please review the code. Thanks

File: examples/Simulations-class-GeneralSimulations.R [hunk 1]
```diff
@@ -1,6 +1,6 @@
 data <- list(
-  Data(x = 1:3, y = 0:2, doseGrid = 1:3, ID = 1L:3L, cohort = 1L:3L),
-  Data(x = 4:6, y = 0:2, doseGrid = 4:6, ID = 1L:3L, cohort = 1L:3L)
+  Data(x = 1:3, y = 0:1, doseGrid = 1:3, ID = 1L:3L, cohort = 1L:3L),
+  Data(x = 4:6, y = 0:1, doseGrid = 4:6, ID = 1L:3L, cohort = 1L:3L)
```
Comment (danielinteractive) [inline examples/Simulations-class-GeneralSimulations.R:3]: if you have 3 doses in `x` you will need 3 outcomes in `y` in each call, e.g. `y = c(0, 1, 0)`

File: R/Simulations-class.R [hunk 2]
```diff
@@ -19,30 +19,35 @@
...
-##' \code{\link{set.seed}} with the \code{seed} as argument before running
-##' \code{\link{simulate,Design-method}}.
-##'
-##' @slot data list of produced \code{\linkS4class{Data}} objects
-##' @slot doses the vector of final dose recommendations
-##' @slot seed random generator state before starting the simulation
-##'
-##' @export
-##' @keywords classes
+# GeneralSimulations class ----
```
Comment (danielinteractive) [inline R/Simulations-class.R:22]:
```suggestion
# GeneralSimulations ----
```

File: R/Simulations-class.R [hunk 3]
```diff
@@ -19,30 +19,35 @@
...
+#'
+#' This class captures trial simulations.
+#' Here also the random generator state before starting the simulation is
+#' saved, in order to be able to reproduce the outcome. For this just use
+#' [`set.seed`] with the `seed` as argument before running
+#' [`simulate,Design-method`].
+#'
+#' @slot data (`list`)\cr list of produced [`Data`] objects
+#' @slot doses (`numeric`)\cr the vector of final dose recommendations
+#' @slot seed (`integer`)\cr random generator state before starting the simulation
```
Comment (danielinteractive) [inline R/Simulations-class.R:38]:
```suggestion
#' @slot data (`list`)\cr produced [`Data`] objects.
#' @slot doses (`numeric`)\cr final dose recommendations.
#' @slot seed (`integer`)\cr random generator state before starting the simulation.
```

File: R/Simulations-class.R [hunk 4]
```diff
@@ -81,17 +86,17 @@
...
+## constructor ----
+
+#' @rdname GeneralSimulations-class
+#'
+#' @param data (`list`)\cr see slot definition.
+#' @param doses (`numeric`)\cr see slot definition.
+#' @param seed (`integer`)\cr see slot definition.
+#'
+#' @example examples/Simulations-class-GeneralSimulations.R
+#' @export
```
Comment (danielinteractive) [inline R/Simulations-class.R:99]: please move the line `# nolint start` from the top of the file to after the code you cleaned up, meaning that linter will apply to the cleaned up area

**Review by danielinteractive — COMMENTED** [review]
Thanks @pengguanya ! Nice work :) Please see few small comments below

**Comment by danielinteractive** [comment]
@pengguanya don't forget to click Update Branch

File: R/Simulations-class.R
Comment (pengguanya) [inline R/Simulations-class.R:1]: @danielinteractive What is the purpose to the curly braces ´{}´ at line 18? Do we need to include other files, or just delete the ´{}´.
I will update line 16 and 17 to remove the redundant ´#´

Comment (danielinteractive) [reply R/Simulations-class.R:1]: You can replace the `{}` by `NULL` to make it more consistent with the other files.
It is needed for `roxygen2` syntax.

File: R/Simulations-class.R [hunk 5]
```diff
@@ -1,4 +1,3 @@
-# nolint start
 #####################################################################################
```
Comment (danielinteractive) [inline R/Simulations-class.R:1]: please also remove these top header lines, we don't need those anymore (lines 1-14)

**Review by danielinteractive — APPROVED** [review]
Thanks @pengguanya !

File: R/Simulations-class.R (hunk 5 above)
Comment (pengguanya) [reply R/Simulations-class.R]: ok, removed

**Pull request merged**